[OSX] Can't finish to build xmr-stak due to Apple Clang Linker

Hi,

Thank your for developing this great xmr miner!
After successful build on Linux, I would like to build on my MacBook Pro Retina 2012 (with Nvidia inside).
My laptop begins to be to old to support CUDA 9, so I use the latest CUDA 8 with Xcode 8.2

`sudo xcode-select -s /Applications/Xcode82.app/Contents/Developer/`
```
cc --version
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin17.0.0
Thread model: posix
InstalledDir: /Applications/Xcode82.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

I choose to use the cmake options in the .travis.yml for Mac : 
`cmake -DMICROHTTPD_ENABLE=OFF -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl  -DOpenCL_ENABLE=OFF ..` in xmr-stak/build folder.
No issue when configuring the build with cmake, then after make clean and make, I've got this issue:
```
Scanning dependencies of target xmr-stak-c
[  3%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_blake256.c.o
[  7%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_groestl.c.o
[ 11%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_jh.c.o
[ 14%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_keccak.c.o
[ 18%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_skein.c.o
[ 22%] Linking C static library bin/libxmr-stak-c.a
[ 22%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-backend
[ 25%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/jconf.cpp.o
[ 29%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/jconf.cpp.o
[ 33%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/minethd.cpp.o
[ 37%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/backendConnector.cpp.o
[ 40%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/globalStates.cpp.o
[ 44%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/crypto/cryptonight_common.cpp.o
[ 48%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/http/httpd.cpp.o
[ 51%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/http/webdesign.cpp.o
[ 55%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/console.cpp.o
[ 59%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/executor.cpp.o
[ 62%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/misc/telemetry.cpp.o
[ 66%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/net/jpsock.cpp.o
[ 70%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/net/socket.cpp.o
[ 74%] Linking CXX static library bin/libxmr-stak-backend.a
/Applications/Xcode82.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: bin/libxmr-stak-backend.a(httpd.cpp.o) has no symbols
/Applications/Xcode82.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: bin/libxmr-stak-backend.a(httpd.cpp.o) has no symbols
[ 74%] Built target xmr-stak-backend
Scanning dependencies of target xmr-stak
[ 77%] Building CXX object CMakeFiles/xmr-stak.dir/xmrstak/cli/cli-miner.cpp.o
[ 81%] Linking CXX executable bin/xmr-stak
[ 81%] Built target xmr-stak
[ 85%] Building NVCC (Device) object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/xmrstak_cuda_backend_generated_cuda_extra.cu.o
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
[ 88%] Building NVCC (Device) object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/xmrstak_cuda_backend_generated_cuda_core.cu.o
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
Scanning dependencies of target xmrstak_cuda_backend
[ 92%] Building CXX object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/jconf.cpp.o
[ 96%] Building CXX object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/minethd.cpp.o
[100%] Linking CXX shared library bin/libxmrstak_cuda_backend.dylib
Undefined symbols for architecture x86_64:
  "win_exit()", referenced from:
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
  "jconf::jconf()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
      xmrstak::nvidia::minethd::self_test() in minethd.cpp.o
  "printer::print_msg(verbosity, char const*, ...)", referenced from:
      xmrstak::nvidia::jconf::parse_config(char const*) in jconf.cpp.o
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::printConfig() in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::generateThreadConfig() in minethd.cpp.o
  "printer::printer()", referenced from:
      xmrstak::nvidia::jconf::parse_config(char const*) in jconf.cpp.o
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::printConfig() in minethd.cpp.o
      printer::inst() in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::generateThreadConfig() in minethd.cpp.o
  "xmrstak::cpu::minethd::func_selector(bool, bool)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "xmrstak::cpu::minethd::thd_setaffinity(_opaque_pthread_t*, unsigned long long)", referenced from:
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
  "xmrstak::cpu::minethd::minethd_alloc_ctx()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "executor::log_result_error(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "executor::executor()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/libxmrstak_cuda_backend.dylib] Error 1
make[1]: *** [CMakeFiles/xmrstak_cuda_backend.dir/all] Error 2
make: *** [all] Error 2
```

I don't know how to resolve this issue ...

Thanks in advance for your help! 👍

I downloaded your zip from your own repo, I use this cmake flags:
`cmake . -DMICROHTTPD_ENABLE=OFF -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DCUDA_ARCH=30 -DOpenCL_ENABLE=OFF -DXMR-STAK_CURRENCY=monero`

and I've got the same issue, sorry ...
```
[100%] Linking CXX shared library bin/libxmrstak_cuda_backend.dylib
Undefined symbols for architecture x86_64:
  "bindMemoryToNUMANode(unsigned long)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "win_exit()", referenced from:
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
  "jconf::GetCurrency()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "jconf::IsCurrencyMonero()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
      _cryptonight_extra_cpu_init in xmrstak_cuda_backend_generated_cuda_extra.cu.o
      _cuda_get_deviceinfo in xmrstak_cuda_backend_generated_cuda_extra.cu.o
  "jconf::jconf()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
      xmrstak::nvidia::minethd::self_test() in minethd.cpp.o
      _cryptonight_extra_cpu_init in xmrstak_cuda_backend_generated_cuda_extra.cu.o
      _cuda_get_deviceinfo in xmrstak_cuda_backend_generated_cuda_extra.cu.o
  "printer::print_msg(verbosity, char const*, ...)", referenced from:
      xmrstak::nvidia::jconf::parse_config(char const*) in jconf.cpp.o
      xmrstak::nvidia::minethd::minethd(xmrstak::miner_work&, unsigned long, xmrstak::nvidia::jconf::thd_cfg const&) in minethd.cpp.o
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::printConfig() in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::generateThreadConfig() in minethd.cpp.o
  "printer::printer()", referenced from:
      xmrstak::nvidia::jconf::parse_config(char const*) in jconf.cpp.o
      xmrstak::nvidia::minethd::minethd(xmrstak::miner_work&, unsigned long, xmrstak::nvidia::jconf::thd_cfg const&) in minethd.cpp.o
      printer::inst() in minethd.cpp.o
      xmrstak::nvidia::minethd::thread_starter(unsigned int, xmrstak::miner_work&) in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::printConfig() in minethd.cpp.o
      xmrstak::nvidia::autoAdjust::generateThreadConfig() in minethd.cpp.o
  "xmrstak::cpu::minethd::func_selector(bool, bool, bool)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "xmrstak::cpu::minethd::thd_setaffinity(_opaque_pthread_t*, unsigned long long)", referenced from:
      xmrstak::nvidia::minethd::minethd(xmrstak::miner_work&, unsigned long, xmrstak::nvidia::jconf::thd_cfg const&) in minethd.cpp.o
  "xmrstak::cpu::minethd::minethd_alloc_ctx()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "xmrstak::strcmp_i(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "executor::log_result_error(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&)", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
  "executor::executor()", referenced from:
      xmrstak::nvidia::minethd::work_main() in minethd.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/libxmrstak_cuda_backend.dylib] Error 1
make[1]: *** [CMakeFiles/xmrstak_cuda_backend.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -429,9 +429,7 @@ if(CUDA_FOUND)
         )
     endif()
     target_link_libraries(xmrstak_cuda_backend ${CUDA_LIBRARIES})
-    if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-        target_link_libraries(xmrstak_cuda_backend xmr-stak-backend)
-    endif()
+    target_link_libraries(xmrstak_cuda_backend xmr-stak-backend)
 endif()
 
 # compile AMD backend
@@ -444,9 +442,7 @@ if(OpenCL_FOUND)
         ${OPENCLSRCFILES}
     )
     target_link_libraries(xmrstak_opencl_backend ${OpenCL_LIBRARY} )
-    if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-        target_link_libraries(xmrstak_opencl_backend xmr-stak-backend)
-    endif()
+    target_link_libraries(xmrstak_opencl_backend xmr-stak-backend)
 endif()
 
 # compile final binary

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -358,6 +358,8 @@ endif()
 
 include_directories(BEFORE .)
 
+set (CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     # remove warnings that f_open() is not save and f_open_s should be used
     add_definitions(-D_CRT_SECURE_NO_DEPRECATE)

